### DISCUSSION TITLE
compile api response too complex

### DISCUSSION THREAD

**Discussion by dassio** [open-policy-agent/community#205] [Community]
from the [blog post](https://blog.openpolicyagent.org/write-policy-in-opa-enforce-policy-in-sql-d9d24db93bf4,)  I was expecting that the partial evaluation would output the simple string, but when using the compile API, the response is an AST which I need to be interpreted and translated for every language (now I am using java), why the compile API is not returning something like the one mentioned in the blog post 
```
# Output #1
pet = data.pets[_]
pet.owner = "alice"
# Output #2
pet = data.pets[_]
pet.veterinarian = "alice";
pet.clinic = "SOMA"
```

**Comment by anderseknert** [comment]
Hi @dassio 👋 

While the example above is certainly more readable than the JSON formatted AST, I think the expectation for the Compile API is that the response will be handled programmatically and not by a human directly. How do you intend to deal with the response?

You could use `opa eval --partial --format source 'data.query.here'` to get something resembling that format (I believe, on the phone 😅) but so far I don't think we've had any requests to surface that format in the REST API.
